buildSrc/src/main/groovy/org/elasticsearch/gradle/test/CommandLineTestTask.groovy

```diff
@@ -0,0 +1,228 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.gradle.test
+
+import org.apache.tools.ant.taskdefs.condition.Os
+import org.elasticsearch.gradle.LoggedExec
+import org.elasticsearch.gradle.VersionProperties
+import org.gradle.api.AntBuilder
+import org.gradle.api.DefaultTask
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.artifacts.Configuration
+import org.gradle.api.plugins.JavaBasePlugin
+import org.gradle.api.tasks.Delete
+import org.gradle.api.tasks.Input
+import org.gradle.util.ConfigureUtil
+
+/**
+ * Starts nodes with different command line options (-p pid, --help, ...) and checks that all is working OK.
+ */
+public class CommandLineTestTask extends DefaultTask {
+
+    ClusterConfiguration clusterConfig = new ClusterConfiguration()
+
+
+    public CommandLineTestTask() {
+
+        description = 'Checks that command line options like -V, -p pid and so on work.'
+        group = JavaBasePlugin.VERIFICATION_GROUP
+
+        // this is our current version distribution configuration we use for all kinds of REST tests etc.
+        project.configurations {
+            elasticsearchDistro
+        }
+        // directory needed for installing and testing
+        File sharedDir = new File(project.buildDir, "cluster/shared")
+        clusterConfig.daemonize = false
+        clusterConfig.distribution = 'zip'
+        NodeInfo node = new NodeInfo(clusterConfig, 0, project, this, VersionProperties.elasticsearch, sharedDir)
+        Configuration configuration = project.configurations.elasticsearchDistro
+
+        // cleanup task
+        Task setup = project.tasks.create(name: ClusterFormationTasks.taskName(this, node, 'clean'), type: Delete) {
+            delete node.homeDir
+            delete node.cwd
+            doLast {
+                node.cwd.mkdirs()
+            }
+        }
+
+        // prepares extraction and configuration setup so we can start the node
+        setup = ClusterFormationTasks.configureExtractTask(ClusterFormationTasks.taskName(this, node, 'extract'), project, setup, node, configuration)
```

Comment:
I might just name the variable for the clean task "clean" and use "setup" for these?